cmd/backup/main.go

```diff
@@ -283,28 +291,43 @@ func (s *script) registerHook(level hookLevel, action func(err error) error) {
 	s.hooks = append(s.hooks, hook{level, action})
 }
 
-// notifyFailure sends a notification about a failed backup run
-func (s *script) notifyFailure(err error) error {
-	body := fmt.Sprintf(
-		"Running docker-volume-backup failed with error: %s\n\nLog output of the failed run was:\n\n%s\n", err, s.output.String(),
-	)
-	title := fmt.Sprintf("Failure running docker-volume-backup at %s", s.start.Format(time.RFC3339))
+// notify sends a notification using the given title and body templates.
+// Automatically adds common params
+func (s *script) notify(titleTemplate string, bodyTemplate string, params map[string]string) error {
+	params["StartTime"] = s.start.Format(time.RFC3339)
+	params["LogOutput"] = s.output.String()
+	params["BackupStopContainerLabel"] = s.c.BackupStopContainerLabel
+
+	titleBuf := &bytes.Buffer{}
+	err := s.template.ExecuteTemplate(titleBuf, titleTemplate, params)
+	if err != nil {
+		return fmt.Errorf("notifyFailure: error executing %s template: %w", titleTemplate, err)
+	}
+	title := strings.Trim(titleBuf.String(), "\n")
+
+	bodyBuf := &bytes.Buffer{}
+	err = s.template.ExecuteTemplate(bodyBuf, bodyTemplate, params)
+	if err != nil {
+		return fmt.Errorf("notifyFailure: error executing %s template: %w", bodyTemplate, err)
+	}
+	body := strings.Trim(bodyBuf.String(), "\n")
```

Comment:
Only 99% sure, but I think Go templates can do this for us if you use '{{- define "foo" -}}`: https://pkg.go.dev/text/template#hdr-Actions

Reply:
Just tested this, and it seems to work when used like this:

```
{{ define "body_failure" -}}
Running docker-volume-backup failed with error: {{ .Error }}

Log output of the failed run was:

{{ .LogOutput }}
{{- end }}
```

Reply:
Oops, I missed this feature. I'll remove the trim and change the default template then.